components/dashboard-modules/EnergyReccomendation.tsx

```diff
@@ -0,0 +1,71 @@
+import Image from 'next/image';
+import { FC } from 'react';
+import { useSiteData } from '~/lib/hooks';
+import { getCurrentTimeForecast } from '~/lib/utils';
+import content from '../../content/power-card-content.json';
+import NumberDisplay from './NumberDisplay';
+
+/**
+ * Determines the appliance with the greatest energy required that is less than or equal to the current output
+ * @param currentOutput the current output in kW based on the getCurrentTimeForecast
+ * @returns the index of the appliance with the greatest enrgy requiement less than or equal to the current ouput
+ * or -1 if an index could not be found
+ */
+
+const getBestRecomendationIndex = (currentOutput: number | undefined) => {
+  if (currentOutput != undefined) {
+    let maxIndex = -1;
+    let maxKW = 0;
+    for (let i = 0; i < content.appliances.length; i++) {
+      const currAppliance = content.appliances[i];
+      if (
+        Number(currAppliance.kW) > maxKW &&
+        Number(currAppliance.kW) <= currentOutput
+      ) {
+        maxIndex = i;
+        maxKW = Number(currAppliance.kW);
+      }
+    }
```

Comment:
this is great, and should work well, but we can also replace this with a `forEach` function! 

```suggestion
    content.appliances.forEach((appliance, i) => {
        if (
          Number(appliance.kW) > maxKW &&
          Number(appliance.kW) <= currentOutput
        ) {
          maxIndex = i;
          maxKW = Number(appliance.kW);
        }
      }
    )
```
something like this should work (the suggestion I gave might be a bit buggy), and getting into the habit of using the `forEach()` function (and other array prototype functions when applicable) will allow our code to be generally a bit less verbose.